### ISSUE TITLE
500 You need to use a client supporting PRET (PRE Transfer) to use PASV

### ISSUE THREAD

**Issue by Hidend** [robinrodricks/FluentFTP#425]
**FTP OS:** Unix

**FTP Server:** ProFTPD

**Computer OS:** Windows

I'm just trying to download a file and I get this error, idk... As an extra info,  HasFeature(FtpCapability.PRET) gives me TRUE but Idk. Check
Capabilities does return this:
MLSD, PRET
With DownloadFileAsync it does work but it doesn't with DownloadFile. I do also get that error listing a directory so I'm not sure anymore

**Logs :**
```
# GetListing("/test/subdir/", Auto)
Command:  TYPE I
Response: 200 Command okay

# OpenPassiveDataStream(AutoPassive, "MLSD /test/subdir", 0)
Command:  EPSV
Response: 502 Command not implemented.

# OpenPassiveDataStream(PASV, "MLSD /test/subdir", 0)
Command:  PASV
Response: 500 You need to use a client supporting PRET (PRE Transfer) to use PASV
```

**Comment by Hidend** [comment]
Using PRET commands manually with .Execute() let me do LIST, RETR etc.

**Comment by robinrodricks** [comment]
Can you paste the code that does not work, and the code that works?

**Comment by Hidend** [comment]
Not working:
```
FtpClient client = new FtpClient("host", 12998, "T", "passwd");
client.DownloadFile("C:\\Users\\Hidend\\Documents\\file.txt", "/main/file.txt");;
client.Disconnect();
Console.ReadKey();
```
```
# Connect()
Status:   Connecting to ***:12998
Response: 220 ProFTPD 1.2.11 Server (GNU/Linux)
Status:   Detected FTP server: ProFTPD
Command:  USER ***
Response: 331 Password required for T
Command:  PASS ***
Response: 230 user logged in successfully.
Response: 230-
Response: 230-
Response: 230-
Command:  FEAT
Response: 211 End
Response: 211-Extensions supported:
Response: MLST type*,x.crc32*,size*,modify*,unix.owner*,unix.group*,x.slaves*,x.xfertime*
Response: PRET
Response: CPSV
Response: NOOP
Response: CLNT
Response: AUTH TLS
Response: PBSZ
Response: SITE XDUPE
Response: SSCN
Status:   Text encoding: System.Text.ASCIIEncoding
Command:  SYST
Response: 215 UNIX system type.
Status:   Auto-detected UNIX listing parser

# DownloadFile("C:\Users\Hidend\Documents\file.txt", "/main/file.txt", Overwrite, None)

# OpenRead("/main/file.txt", Binary, 0)
Command:  TYPE I
Response: 200 Command okay

# OpenPassiveDataStream(AutoPassive, "RETR /main/file.txt", 0)
Command:  EPSV
Response: 502 Command not implemented.

# OpenPassiveDataStream(PASV, "RETR /main/file.txt", 0)
Command:  PASV
Response: 500 You need to use a client supporting PRET (PRE Transfer) to use PASV
```

Working:
```
FtpClient client = new FtpClient("host", 12998, "T", "passwd");
client.Execute("PRET RETR /main/file.txt");
client.DownloadFile("C:\\Users\\Hidend\\Documents\\file.txt", "/main/file.txt"); ;
client.Disconnect();
Console.ReadKey();
```
```

# Connect()
Status:   Connecting to ***:12998
Response: 220 ProFTPD 1.2.11 Server (GNU/Linux)
Status:   Detected FTP server: ProFTPD
Command:  USER ***
Response: 331 Password required for T.
Command:  PASS ***
Response: 230 user logged in successfully.
Response: 230-
Response: 230-
Response: 230-
Command:  FEAT
Response: 211 End
Response: 211-Extensions supported:
Response: MLST type*,x.crc32*,size*,modify*,unix.owner*,unix.group*,x.slaves*,x.xfertime*
Response: PRET
Response: CPSV
Response: NOOP
Response: CLNT
Response: AUTH TLS
Response: PBSZ
Response: SITE XDUPE
Response: SSCN
Status:   Text encoding: System.Text.ASCIIEncoding
Command:  SYST
Response: 215 UNIX system type.
Status:   Auto-detected UNIX listing parser
Command:  PRET RETR /main/file.txt
Response: 200 OK, planning for upcoming download

# DownloadFile("C:\Users\Hidend\Documents\file.txt", "/main/file.txt", Overwrite, None)

# OpenRead("/main/file.txt", Binary, 0)
Command:  TYPE I
Response: 200 Command okay

# OpenPassiveDataStream(AutoPassive, "RETR /main/file.txt", 0)
Command:  EPSV
Response: 502 Command not implemented.

# OpenPassiveDataStream(PASV, "RETR /main/file.txt", 0)
Command:  PASV
Response: 227 Entering Passive Mode (54,38,178,128,56,18).
Response: 227- Using SLAVE1 for upcoming transfer
Status:   Connecting to ***:14354
Command:  RETR /main/file.txt
Response: 150 File status okay; about to open data connection from SLAVE1.
Status:   Disposing FtpSocketStream...
Response: 226 Transfer complete, 68B in 0.0 seconds (0B/s)
Response: 226- Checksum from transfer: 860215e3
Command:  QUIT
Response: 221 Goodbye!
Status:   Disposing FtpSocketStream...
```

**Comment by robinrodricks** [comment]
I'll try but there is this post on the [Filezilla Client forum](https://forum.filezilla-project.org/viewtopic.php?t=17085) where they say PRET is weird and has no proper spec. Still, I'll try.

**Comment by Hidend** [comment]
Since it says:
Supports DrFTPD's PRET command and in [FTP Support](https://github.com/robinrodricks/FluentFTP#ftp-support) was listed and said "Automatic", I thought it was supported haha.

Ok thanks dude!

**Comment by robinrodricks** [comment]
That was written by the original author years ago. He apparently didn't add any support for it :(

**Comment by robinrodricks** [comment]
Please try this...

If the server supports PRET we send it before transferring data...

https://we.tl/t-A1RiuFqlzF

**Comment by Hidend** [comment]
Downloading works, thank you!

**Comment by robinrodricks** [comment]
Released in 26.0.0.

https://www.nuget.org/packages/FluentFTP/26.0.0
